src/wp-includes/template.php

```diff
@@ -729,15 +729,18 @@ function locate_template( $template_names, $load = false, $load_once = true, $ar
 		if ( ! $template_name ) {
 			continue;
 		}
-		if ( file_exists( $wp_stylesheet_path . '/' . $template_name ) ) {
-			$located = $wp_stylesheet_path . '/' . $template_name;
+		$stylesheet      = $wp_stylesheet_path . '/' . $template_name;
+		$template        = $wp_template_path . '/' . $template_name;
+		$template_compat = ABSPATH . WPINC . '/theme-compat/' . $template_name;
+
+		if ( file_exists( $stylesheet ) && 0 === validate_file( $stylesheet ) ) {
+			$located = $stylesheet;
 			break;
-		} elseif ( $is_child_theme && file_exists( $wp_template_path . '/' . $template_name ) ) {
+		} elseif ( $is_child_theme && file_exists( $template ) && 0 === validate_file( $template ) ) {
 			$located = $wp_template_path . '/' . $template_name;
```

Comment:
This could be simplified a bit.
```suggestion
			$located = $template;
```

Reply:
@martinkrcho Thanks for the feedback. I have added suggested changes in latest commit.

Reply:
Thanks.